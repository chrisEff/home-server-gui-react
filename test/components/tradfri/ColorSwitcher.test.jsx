const React = require('react')
const ColorSwitcher = require('../../../src/components/tradfri/ColorSwitcher')

const { shallow } = require('enzyme')

describe('ColorSwitcher', () => {

	it('should render the rgb switcher correctly', () => {
		const wrapper = shallow(<ColorSwitcher bulbType="rgb" color="red" />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the white-spectrum switcher correctly', () => {
		const wrapper = shallow(<ColorSwitcher bulbType="white-spectrum" color="warm" />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the white switcher correctly', () => {
		const wrapper = shallow(<ColorSwitcher bulbType="white" color="neutral" />)
		expect(wrapper).toMatchSnapshot()
	})

})
