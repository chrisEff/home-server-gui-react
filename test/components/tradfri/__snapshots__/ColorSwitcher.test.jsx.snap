// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSwitcher should render the rgb switcher correctly 1`] = `
<div
  className="color-switcher"
>
  <div
    className="color red active"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color green"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color blue"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color yellow"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color pink"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color purple"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color warm"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color neutral"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color cold"
    onClick={[Function]}
  >
    <div />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white switcher correctly 1`] = `
<div
  className="color-switcher white"
>
  <div
    className="color neutral active"
  >
    <div />
  </div>
</div>
`;

exports[`ColorSwitcher should render the white-spectrum switcher correctly 1`] = `
<div
  className="color-switcher spectrum"
>
  <div
    className="color warm active"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color neutral"
    onClick={[Function]}
  >
    <div />
  </div>
  <div
    className="color cold"
    onClick={[Function]}
  >
    <div />
  </div>
</div>
`;
